### PR TITLE
fix: network policy update

### LINKED ISSUE #2593: Network Policy SDK update
### Use Cases or Problem Statement

Follow up for https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2514. Because there were new fields added to the documentation, we have to adjust the current implementation in the SDK to support those new fields.

### How much impact is this issue causing?

Medium

### Additional Information

Resource implementation reference: https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2482

### PR DESCRIPTION
**Pull request by sfc-gh-swinkler**
fixes #2593 

## Test Plan

* [ ] acceptance tests

* [ ] …

## References

*

### CODE REVIEW

File: pkg/sdk/testint/network_policies_gen_integration_test.go [hunk 1]
```diff
@@ -85,6 +108,90 @@ func TestInt_NetworkPolicies(t *testing.T) {
...
+	t.Run("Alter - set allowed network rule list", func(t *testing.T) {
+		allowedNetworkRule := createNetworkRuleHandle(t, client)
+
+		req := defaultCreateRequest()
+		err, dropNetworkPolicy := createNetworkPolicy(t, client, req)
+		require.NoError(t, err)
+		t.Cleanup(dropNetworkPolicy)
+
+		err = client.NetworkPolicies.Alter(ctx, sdk.NewAlterNetworkPolicyRequest(req.GetName()).
+			WithSet(sdk.NewNetworkPolicySetRequest().WithAllowedNetworkRuleList([]sdk.SchemaObjectIdentifier{allowedNetworkRule})))
```
Comment (sfc-gh-jcieslak) [inline pkg/sdk/testint/network_policies_gen_integration_test.go:120]: can we add/extend this test to have two items? We'll know the SQL is generating correctly with commas and we're testing that it works on the Snowflake side.

File: pkg/sdk/network_policies_def.go [hunk 2]
```diff
@@ -35,16 +47,28 @@ var (
...
+						ListAssignment("BLOCKED_NETWORK_RULE_LIST", "SchemaObjectIdentifier", g.ParameterOptions().Parentheses()).
 						ListQueryStructField("AllowedIpList", ip, g.ParameterOptions().SQL("ALLOWED_IP_LIST").Parentheses()).
 						ListQueryStructField("BlockedIpList", ip, g.ParameterOptions().SQL("BLOCKED_IP_LIST").Parentheses()).
 						OptionalTextAssignment("COMMENT", g.ParameterOptions().SingleQuotes()).
-						WithValidation(g.AtLeastOneValueSet, "AllowedIpList", "BlockedIpList", "Comment"),
+						WithValidation(g.AtLeastOneValueSet, "AllowedIpList", "BlockedIpList", "Comment", "AllowedNetworkRuleList", "BlockedNetworkRuleList"),
 					g.KeywordOptions().SQL("SET"),
 				).
+				OptionalQueryStructField(
+					"Add",
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/network_policies_def.go:59]: In the docs ADD and REMOVE both take only one network rule and not the list like CREATE/ALTER SET. Is this the error in the docs or in the implementation?

File: pkg/sdk/testint/network_policies_gen_integration_test.go [hunk 3]
```diff
@@ -85,6 +108,90 @@ func TestInt_NetworkPolicies(t *testing.T) {
 		assert.Equal(t, 1, np.EntriesInBlockedIpList)
 	})
 
+	t.Run("Alter - set allowed network rule list", func(t *testing.T) {
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/testint/network_policies_gen_integration_test.go:111]: This test is failing on the CI:` FAIL: TestInt_NetworkPolicies/Alter_-_set_allowed_network_rule_list (2.05s)`.

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]

**Review by sfc-gh-jcieslak — DISMISSED** [review]

File: pkg/sdk/network_policies_def.go (hunk 2 above)
Comment (sfc-gh-swinkler) [reply pkg/sdk/network_policies_def.go:59]: the docs are wrong, it should be a list, i tried one network rule, and it failed.

File: pkg/sdk/testint/network_policies_gen_integration_test.go (hunk 3 above)
Comment (sfc-gh-swinkler) [reply pkg/sdk/testint/network_policies_gen_integration_test.go:111]: fixed

File: pkg/sdk/network_policies_def.go (hunk 2 above)
Comment (sfc-gh-asawicki) [reply pkg/sdk/network_policies_def.go:59]: Please add it to the doc discuss channel, and then you can close this comment

**Review by sfc-gh-asawicki — APPROVED** [review]

**Pull request merged**